client/src/components/App/Content/InvestigationForm/InvestigationForm.tsx

```diff
@@ -81,39 +68,66 @@ const InvestigationForm: React.FC = (): JSX.Element => {
             setSymptomsStartDate, setExposureDate, setHasSymptoms, setEndInvestigationDate]
     );
 
-    const { currentTab, confirmFinishInvestigation, handleSwitchTab, saveCurrentTab, isButtonDisabled, areThereContacts } = useInvestigationForm({ clinicalDetailsVariables, personalInfoData, exposuresAndFlightsVariables, interactedContactsState });
-    
-    const shouldDisableButton = isButtonDisabled(currentTab.name);
-    const isLastTab = (currentTab.id === LAST_TAB_ID || (currentTab.id === LAST_TAB_ID - 1 && !areThereContacts));
-    
+    const { confirmFinishInvestigation, areThereContacts } = useInvestigationForm();
+    const {
+        currentTab,
+        moveToNextTab,
+        setNextTab
+    } = useTabManagement();
+
+    const isInvestigationValid = () => {
+        return !(formsValidations.some((formValidation) => !formValidation));
+    }
+
+    const handleClick = () => {
+        if(currentTab === LAST_TAB_ID) { 
+            if(isInvestigationValid()) {
+                confirmFinishInvestigation(epidemiologyNumber);
+            } else {
+                Swal.fire({
+                    icon: 'error',
+                    title: 'חלק מן השדות אינם תקניים, נא מלא אותם מחדש ונסה שוב.',
+                    customClass: {
+                        title:classes.swalTitle
+                    },
+                });
+            }
+        } else {
+            setNextTab(currentTab + 1);
+        }
+    }
+
+    const getLastTabId = () => {
+        return areThereContacts ? LAST_TAB_ID : LAST_TAB_ID - 1;   
+    }
```

Comment:
you don't need a function, can use a variable instead
const lastTabId = areThereContacts ? LAST_TAB_ID : LAST_TAB_ID - 1;